beets/plugins.py

```diff
@@ -478,9 +478,39 @@ def sanitize_choices(choices, choices_all):
     others = [x for x in choices_all if x not in choices]
     res = []
     for s in choices:
-        if s in list(choices_all) + ['*']:
-            if not (s in seen or seen.add(s)):
-                res.extend(list(others) if s == '*' else [s])
+        if s not in seen:
+            if s in list(choices_all):
+                res.append(s)
+            elif s == '*':
+                res.extend(others)
+        seen.add(s)
+    return res
+
+
+def sanitize_pairs(pairs, pairs_all):
+    """Clean up a single-element mapping configuration attribute: keep only
+    choices elements present in pairs_all, remove duplicate elements, expand
+    ('str', '*') and ('*', '*') wildcards while the keeping original order.
```

Comment:
Maybe a quick example would help describe what's going on here? This seems like a useful utility (and a useful configuration option format), and other plugins in the future might want to use it.

Reply:
An example might be appropriate, but where would I put it? In the [confuse docs](http://confuse.readthedocs.io/en/latest/#validation), an `as_pairs` method is actually documented (which is only very slightly different). There appeared to be no code for it, however (I did not check the history, it might have existed before). So either an inline comment, or a cherry-pick from the confuse repo and an addition to its documentation. I don't think I'm aware of the status of moving confit/confuse out of this repository, so I'm not quite sure which to prefer.